core/test-utils/src/e2e/node.rs

```diff
@@ -1,104 +1,325 @@
+use std::any::Any;
 use std::net::SocketAddr;
 use std::path::PathBuf;
+use std::time::Duration;
 
+use anyhow::Result;
+use fdi::MultiThreadedProvider;
 use fleek_crypto::{
+    AccountOwnerPublicKey,
     AccountOwnerSecretKey,
+    ConsensusPublicKey,
     ConsensusSecretKey,
     EthAddress,
+    NodePublicKey,
     NodeSecretKey,
     SecretKey,
 };
-use lightning_application::state::QueryRunner as ApplicationQuery;
+use lightning_application::state::QueryRunner;
 use lightning_application::Application;
-use lightning_broadcast::Broadcast;
 use lightning_checkpointer::Checkpointer;
 use lightning_interfaces::prelude::*;
-use lightning_node::Node;
+use lightning_node::ContainedNode;
 use lightning_notifier::Notifier;
 use lightning_pool::PoolProvider;
 use lightning_rep_collector::MyReputationReporter;
 use lightning_rpc::Rpc;
 use lightning_signer::Signer;
 use lightning_utils::transaction::TransactionSigner;
-use ready::tokio::TokioReadyWaiter;
-use types::{NodeIndex, NodeInfo};
+use merklize::StateRootHash;
+use types::{
+    Epoch,
+    ExecuteTransactionError,
+    ExecuteTransactionOptions,
+    ExecuteTransactionRequest,
+    ExecuteTransactionResponse,
+    ExecuteTransactionWait,
+    Genesis,
+    NodeIndex,
+    NodeInfo,
+    NodePorts,
+    UpdateMethod,
+};
 
-use super::TestNodeComponents;
+use super::SyncBroadcaster;
 use crate::consensus::MockForwarder;
 use crate::keys::EphemeralKeystore;
 
-pub struct TestNode {
-    pub inner: Node<TestNodeComponents>,
-    pub before_genesis_ready: TokioReadyWaiter<TestNodeBeforeGenesisReadyState>,
-    pub after_genesis_ready: TokioReadyWaiter<()>,
+#[async_trait::async_trait]
+pub trait TestNetworkNode {
+    fn as_any(&self) -> &dyn Any;
+    fn index(&self) -> NodeIndex;
+    async fn start(&mut self) -> Result<()>;
+    async fn shutdown(self: Box<Self>);
+    async fn apply_genesis(&self, genesis: Genesis) -> Result<()>;
```

Comment:
Note that we can remove async trait completely (and the latency related to boxed dynamics) with a slightly more verbose syntax like so:
```suggestion
    fn start(&mut self) -> impl Future<Output = Result<()>> + Send;
    fn shutdown(self: Box<Self>) -> impl Future<Output = ()> + Send;
    fn apply_genesis(&self, genesis: Genesis) -> impl Future<Output = Result<()>> + Send;
```

No modifications are needed by trait implementers due to how the desugaring works. I know it's not too crucial as it's just a testing framework, but we already switched all the component traits to this pattern

Reply:
I haven't been able to get this to work because it results in it not being object-safe and so would need to then box/pin them to fix that, which ends up looking a lot like what async_trait is doing anyway.